007_team_5_agent_design/007_team_5_agent_design.tex

```diff
@@ -1 +1,522 @@
+
+Section \ref{sec:team5-overview} provides an overview of the agent's parameters and strategy, Section \ref{sec:team5-messaging} describes the agent's communication strategy, Section \ref{sec:team5-memory} describes how the agent stores its social network and how it judges other agents, and Section \ref{sec:team5-treaties} describes how the Team 5 agent utilises treaties. Finally, Section \ref{sec:team5-experimentation} contains analysis of this agent including how it performs in a Tower consisting of itself alone as well as together with other agent types and Section \ref{sec:team5-conclusion} provides summarising remarks regarding the implementation of this agent type.
+
+\section{Agent Overview}
+\label{sec:team5-overview}
+
+\ToDo{Give better description of Run function}
+
+The strategy employed by this agent aims to maximise its chances of survival while also considering those who have helped the agent in the past.
+
+At the beginning of each day, the agent's `personality' is updated using various parameters stored from previous days. These personality parameters are shown in Table \ref{tab:team5-personality}.
+
+\begin{table}
+    \centering
+\begin{tabular}%
+    {| >{\raggedleft\arraybackslash}p{0.25\linewidth} | %
+    >{\raggedright\arraybackslash}p{0.65\linewidth} | %
+    }
+    \hline
+    Selfishness          & Ranges from 0 to 10. Determines how much the agent will act for their own survival as opposed to the best interests of the surrounding agents\\
+    \hline
+    Last Meal            & How much the agent ate in their last meal\\
+    \hline
+    Days Since Last Meal & How many days since the agent last ate\\
+    \hline
+    % HP After Eating      & \\
+    % \hline
+    Aim HP               & Defines the goal HP the agent would like to reach at the end of the day\\
+    \hline
+    Attempt Food         & The amount of food the agent will attempt to take from the platform\\
+    \hline
+    Messaging Counter    & Determines which message type to send to which floor \\
+    \hline
+    % Treaty Send Counter  & \\
+    % \hline
+    % Attempt To Eat       & \\
+    % \hline
+    Leadership           & A threshold value required for the agent to propose treaties\\
+    \hline
+    Social Memory        & Stores the information the agent learns about other agents, as well as its opinion of them (see Table \ref{tab:team5-memory})\\
+    \hline
+    Surrounding Agents   & A list of the agent's neighbours\\
+    \hline
+\end{tabular}
+\caption{The personality parameters of Team 5's agent}
+\label{tab:team5-personality}
+\end{table}
+
+At the start of each tick, the agent checks to see if it has received any messages and if so, will respond to these\footnote{Note that an agent can receive and respond to only one message per tick.}. The agent then sends out messages to other agents. The mechanism for messaging is described in Section \ref{sec:team5-messaging}.
```

Comment:
```suggestion
Each of these personality parameters are linked, with each of them contributing to the agent deciding how much food it will attempt to eat that day. This is explained in further detail in sections to follow.

At the start of each tick, the agent checks to see if it has received any messages and if so, will respond to these\footnote{Note that an agent can receive and respond to only one message per tick.}. The agent then sends out messages to other agents. The mechanism for messaging is described in Section \ref{sec:team5-messaging}.
```